scripts/west_commands/build.py

```diff
@@ -396,7 +397,8 @@ def _run_cmake(self, board, origin, cmake_opts):
         # to Just Work:
         #
         # west build -- -DOVERLAY_CONFIG=relative-path.conf
-        final_cmake_args = ['-B{}'.format(self.build_dir),
+        final_cmake_args = ['-DWEST_PYTHON={}'.format(sys.executable),
```

Comment:
Shouldn't we have this in scripts/west_commands/zcmake.py's run_cmake() instead? Is there a reason not to have it there?

Reply:
that's a very good suggestion.
Then we have this piece of information for all cmake invocations.

Will move.

Reply:
done

Reply:
Moved the code back here.

The zcmake.py's run_cmake() method is a generic method that can be used both with `--build`, `-P` (script mode, as here: https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/west_commands/export.py#L68-L69), and for CMake build configure and generator, like we do here.

In fact we only want this flag when generating build files for Zephyr, so it kind of belong together with out `-G` flag.
Should have thought of that before blindly moving the code in the first place.